### PR TITLE
feat: [IOPLT-342] Bump react-native version to 0.70

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR bumps the `react-native` version to 0.70

## List of changes proposed in this pull request
- Upgrades rn version
- Upgrade native files
- upgrade code reference where needed

## How to test
```shell
cd android && ./gradlew clean && cd..
rm -rf node-modules
rm -rf ios/Pods
rm -rf ios/DerivedData
rm -rf ios/build


yarn install
pod install
```
then run on both ios and android

### CODE REVIEW

File: android/app/src/main/jni/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.13)
+
+# Define the library name here.
+project(rndiffapp_appmodules)
```
Comment (CrisTofani) [inline android/app/src/main/jni/CMakeLists.txt:4]:
```suggestion
project(italiaapp_appmodules)
```

File: android/app/src/main/jni/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.13)
+
+# Define the library name here.
+project(italiaapp_appmodules)
```
Comment (CrisTofani) [inline android/app/src/main/jni/CMakeLists.txt:4]:
```suggestion
project(ioapp_appmodules)
```

**Comment by CrisTofani** [comment]
> From a static perspective everything looks good even by double checking with the React Native Upgrade Helper. Just a side note, should we update the `patches.md` file accordingly?

Thank you @LazyAfternoons , done in [1941f94](https://github.com/pagopa/io-app/pull/5506/commits/1941f94c8563d08d3d7ab14bdab9768d1af56c64)

**Review by dmnplb — APPROVED** [review]

**Review by LazyAfternoons — APPROVED** [review]
LGTM on behalf of FCI. 
Checked with [React Native Upgrade Helper]( https://react-native-community.github.io/upgrade-helper/) from a static perspective.
I tested a signing flow with the local `env` on iPhone 14 simulator@iOS 16.4.
I tested a signing flow with production `env` on iPhone XS@iOS 16.4 and Pixel 4a@Android 13.

**Review by hevelius — APPROVED** [review]
LGTM. Tested on physical devices (both iOS and Android) on:
* FCI signing flow
* Wallet
* Payment flow

**Comment by dmnplb** [comment]
@shadowsheep1 @Ladirico Before merging, try to check the authentication flows, as I've noticed some rendering bugs while working on the `native-base' removal branch (especially in the email validation flow)

**Review by shadowsheep1 — APPROVED** [review]
LGTM 👍. @dmnplb I've tested the email validation flow and found no issues.

**Pull request merged**